app/services/catalog/cancel_order.rb

```diff
@@ -17,6 +17,9 @@ def process
       end
 
       self
+    rescue Catalog::ApprovalError => e
```

Comment:
@eclarizio were we able to find out why in every case calling `/cancel` was raising the 500 as opposed to getting caught at line 11?